docs/auth/jwt_auth.rst

```diff
@@ -0,0 +1,237 @@
+There is no right or wrong method to work out when to expire JWT, but for arguments sake let's say we want them to last a week.
+I also like to set my tokens to expire at a time when they are unlikely to be being used, to save a user potentially getting a broken session
+
+.. code-block:: python
+
+    def set_token_expiry():
+        ''' We want to set the token expiry to be 7 days
+            but at 3am, this should stop users been thrown out
+            of a session during normal working time
+        '''
+        now = datetime.datetime.now()
+        expiry = now + datetime.timedelta(days=8)
+        expiry_midnight = expiry.replace(hour=3, minute=0, second=0)
+        # Calculate the difference in seconds between two dates
+        seconds_to_expire = (expiry_midnight-now).total_seconds()
+        return seconds_to_expire
+
+
+
+Now we have all of the information we need to create a basic JWT token, there are more examples of this online, but the below covers us for now
+
+.. code-block:: python
+
+    def create_token(key, user_id, permissions, user_name, expiration, audience=None):
+        # create an empty dict
+        payload = {}
+        # set the expiration
+        payload["exp"] = expiration
+        if audience:
+            # audience claims are out of scope for this tutorial, but
+            # a useful placeholder to expand this code
+            payload["aud"] = audience
+        # 'sub' is the primary contents of a JWT, so I user "user_id" here
+        payload['sub'] = user_id
+        # available permissions associated to this user or JWT, not required here
+        # but helpful to explain how they work
+        payload['permissions'] = permissions
+        # again an extension of the token, we will explain further on why only user_id
+        # is required, but passing back to the frontend can be very useful.
+        payload['user_name'] = user_name
+        token = jwt.encode(payload, key, algorithm=JWT_ALGO, json_encoder=None)
+        if not isinstance(token, str):  # Python3 unicode madness
+            token = token.decode("ascii")
+        return token
+
+Now the token should look something like
+
+.. code-block:: text
+
+    eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJleHAiOjY1MDM5Ny4wLCJzdWIiOjk5OSwicGVybWlzc2lvbnMiOlsidmlldyIsImVkaXQiXSwidXNlcl9uYW1lIjoiam9lIn0.dTCQxCorZtzIExeUtxB49_JAKljS9M8FZEhBvT_JPudzGuOyTPGYpTaxgaYWEjvnMG1m_kKvASJcn77Klgb9lQ
+
+We can work onto the next phase, intercepting this token in the security policy and evaluating it.
+It is important to note that all data inside a JWT is available for anyone to view it.
+All the secret does is verify the token has not been manipulated, so NEVER send information inside a JWT that you don't mind getting out.
+
+
+Evaluating with Pyramid 
+#######################
+
+Now we have sent the JWT token to our front-end application, we can presume that it will get sent back to the Pyramid application 
```

Comment:
```suggestion
Now that we have sent the JWT token to our front-end application, we can presume that it will get sent back to the Pyramid application 
```